packages/engine/Source/Scene/Scene.js

```diff
@@ -1787,69 +1792,66 @@ function updateDerivedCommands(scene, command, shadowsDirty) {
  * @private
  */
 Scene.prototype.updateDerivedCommands = function (command) {
-  if (!defined(command.derivedCommands)) {
+  const { derivedCommands } = command;
+  if (!defined(derivedCommands)) {
     // Is not a DrawCommand
     return;
   }
 
-  const frameState = this._frameState;
+  const { shadowState, useLogDepth } = this._frameState;
   const context = this._context;
 
   // Update derived commands when any shadow maps become dirty
   let shadowsDirty = false;
-  const lastDirtyTime = frameState.shadowState.lastDirtyTime;
+  const lastDirtyTime = shadowState.lastDirtyTime;
   if (command.lastDirtyTime !== lastDirtyTime) {
     command.lastDirtyTime = lastDirtyTime;
     command.dirty = true;
     shadowsDirty = true;
   }
 
-  const useLogDepth = frameState.useLogDepth;
   const useHdr = this._hdr;
-  const derivedCommands = command.derivedCommands;
   const hasLogDepthDerivedCommands = defined(derivedCommands.logDepth);
   const hasHdrCommands = defined(derivedCommands.hdr);
   const hasDerivedCommands = defined(derivedCommands.originalCommand);
   const needsLogDepthDerivedCommands =
     useLogDepth && !hasLogDepthDerivedCommands;
   const needsHdrCommands = useHdr && !hasHdrCommands;
+  // PERFORMANCE_IDEA: if (!useLogDepth && !useHdr), we may be able to save a call to updateDerivedCommands.
```

Comment:
This may be up for debate, but I'm generally not a fan of leaving TODO comments like this in the code. It's too easy to forget about them and then they never get implemented. Do you think this will indeed be useful for a future developer?